src/Client.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Http\Client;
+
+/**
+ * Interface for both HTTP Client type
+ *
+ * @author Márk Sági-Kazár mark.sagikazar@gmail.com>
+ */
+interface Client extends PsrHttpClient, HttpClient
```

Comment:
Well, this interface should rather be implemented by actual HTTP Clients, like guzzle. Also eases implementing both interfaces extended by this one. Probably this is only for type hint convenience. The naming is on the TODO list, because I am not sure about the current state either. The reason why I named HttpClient this way is that it is clearly about HTTP methods. Probably the word "method" could be used somehow in the interface? I am open for ideas.

BTW I agree about the documentation should clear all those questions that originates from naming.